src/apply_rules/property_rule.rs

```diff
@@ -883,6 +894,123 @@ fn update_stop_points_equipments(
     }
 }
 
+fn get_id_or_create_trip_property(
+    trip_property: TripProperty,
+    collection_trip_properties: &mut CollectionWithId<TripProperty>,
+    prefix: &str,
+) -> String {
+    fn generate_trip_property_id(
+        collection_trip_properties: &mut CollectionWithId<TripProperty>,
+        prefix: &str,
+    ) -> String {
+        let mut inc = 0;
+        let mut available = false;
+        let mut trip_property_id = String::new();
+        while !available {
+            inc += 1;
+            trip_property_id = format!("{}{}", prefix, inc);
+            if collection_trip_properties.get(&trip_property_id).is_none() {
+                available = true;
+            }
+        }
+        trip_property_id
+    }
+
+    let similar_trip_properties: Vec<&TripProperty> = collection_trip_properties
+        .values()
+        .filter(|tp| tp.is_similar(&trip_property))
+        .collect();
+
+    if !similar_trip_properties.is_empty() {
+        // Similar trip_properties may exist, especially after a merge of ntfs (with different prefixes)
+        // We take the first one that matches. Rare case.
+        return similar_trip_properties[0].id.clone();
+    }
+
+    let trip_property_id = generate_trip_property_id(collection_trip_properties, prefix);
+
+    collection_trip_properties
+        .push(TripProperty {
+            id: trip_property_id.clone(),
+            wheelchair_accessible: trip_property.wheelchair_accessible,
+            bike_accepted: trip_property.bike_accepted,
+            air_conditioned: trip_property.air_conditioned,
+            visual_announcement: trip_property.visual_announcement,
+            audible_announcement: trip_property.audible_announcement,
+            appropriate_escort: trip_property.appropriate_escort,
+            appropriate_signage: trip_property.appropriate_signage,
+            school_vehicle_type: trip_property.school_vehicle_type,
+        })
```

Comment:
This can be written like this.

```rust
.push(TripProperty {
    id: trip_property_id.clone(),
    ..trip_property
})
```

Another possibility is to give `mut` to `trip_property` and then do this.

```rust
trip_property.id = trip_property_id.clone();
collection_trip_properties.push(trip_property).unwrap();
```

Reply:
I opted for the 1st solution :+1: